rust/ommx/src/instance.rs

```diff
@@ -50,3 +50,239 @@ pub struct Instance {
     #[getset(get = "pub")]
     constraint_hints: ConstraintHints,
 }
+
+impl Instance {
+    ///
+    /// - All `VariableID`s in `Function`s contained both directly and indirectly must be keys of `decision_variables`.
+    /// - Key of `constraints` and `removed_constraints` are disjoint.
+    ///
+    pub fn new(
+        sense: Sense,
+        objective: Function,
+        decision_variables: BTreeMap<VariableID, DecisionVariable>,
+        constraints: BTreeMap<ConstraintID, Constraint>,
+        removed_constraints: BTreeMap<ConstraintID, RemovedConstraint>,
```

Comment:
`removed_constraints` should not be added here. It should be set via `relax,restore_constratins`